agent/structs/check_type.go

```diff
@@ -91,6 +93,8 @@ func (t *CheckType) UnmarshalJSON(data []byte) (err error) {
 	}{
 		Alias: (*Alias)(t),
 	}
+	aux.H2PingUseTLS = true
```

Comment:
I set the defaults here and in https://github.com/hashicorp/consul/pull/10690/files#diff-ecefd48ddc3a89af6c4b790a7d26ebf1867caaffcf164fdc350eb899f70dd0c2R80. I don't know if this is the best way to do it, but I could not find another example of this particular scenario (where a deliberately set false value must be distinguished from an empty struct value). Please let me know if there is a better way. I also added the default here https://github.com/hashicorp/consul/pull/10690/files#diff-77452bdff6ee872bdd7b08a2603dfb39d165d45c5bcafc5f8bccaaf010aa7e85R1574, but was wondering where this code is called because it did not seem to make a difference for the scenarios I was manually testing.